synapseclient/client.py

```diff
@@ -3124,14 +3125,17 @@ def _async_table_update(self, table, changes=[], create_snapshot=False,
 
         :return:  Snapshot Response
         """
-
+        snapshot_options = {'snapshotComment': comment,
+                            'snapshotLabel': label,
+                            'snapshotActivityId': activity}
+        new_snapshot = {key:value for key, value in snapshot_options.items() if value}
         table_update_body = {'changes': changes,
                              'createSnapshot': create_snapshot,
-                             'snapshotOptions': {'snapshotComment': comment,
-                                                 'snapshotLabel': label,
-                                                 'snapshotActivityId': activity}}
+                             'snapshotOptions': new_snapshot}
+        new_table_update_body = {key:value for key, value in table_update_body.items() if value}
```

Comment:
I think none of these are optional so we don't need to prune for this https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/table/TableUpdateTransactionRequest.html